fhir-persistence-blob-app/src/main/java/com/ibm/fhir/persistence/blob/app/PayloadReconciliation.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * (C) Copyright IBM Corp. 2022
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+ 
+package com.ibm.fhir.persistence.blob.app;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.azure.core.http.rest.Response;
+import com.ibm.fhir.config.FHIRRequestContext;
+import com.ibm.fhir.database.utils.api.ITransaction;
+import com.ibm.fhir.database.utils.model.DbType;
+import com.ibm.fhir.database.utils.pool.DatabaseSupport;
+import com.ibm.fhir.persistence.blob.BlobContainerManager;
+import com.ibm.fhir.persistence.blob.BlobDeletePayload;
+import com.ibm.fhir.persistence.blob.BlobManagedContainer;
+import com.ibm.fhir.persistence.blob.BlobResourceScanner;
+import com.ibm.fhir.persistence.exception.FHIRPersistenceException;
+import com.ibm.fhir.persistence.jdbc.dao.api.ResourceRecord;
+import com.ibm.fhir.persistence.jdbc.dao.impl.ResourceExistsDAO;
+
+/**
+ * Implements an algorithm to scan the offload persistence store and check
+ * that the RDBMS contains the corresponding record. As the RDBMS is the
+ * source of truth, any records in the persistence store without a
+ * corresponding record in the RDBMS should be deleted.
+ */
+public class PayloadReconciliation {
+    private static final Logger logger = Logger.getLogger(PayloadReconciliation.class.getName());
+
+    // The tenant id to use in the request context
+    private final String tenantId;
+
+    // The tenant's datastore (usually "default")
+    private final String dsId;
+
+    // Scan and report but don't delete
+    private final boolean dryRun;
+
+    // Provides access to database connections and transaction handling
+    private final DatabaseSupport dbSupport;
+
+    // Tracking how many resource versions we process
+    private long totalProcessed = 0;
+    
+    // Simple map cache of resource type id to name
+    private final Map<Integer,String> resourceTypeMap = new HashMap<>();
+
+    // Stop scanning after we hit this number of seconds
+    private final int maxScanSeconds;
+    /**
+     * Public constructor
+     * @param tenantId
+     * @param dsId
+     * @param dbProperties
+     * @param dbType
+     * @param dryRun
+     */
```

Comment:
javadoc: Several methods in this class have params not matching those method's javadoc params.

Reply:
fixed